Tribler/Core/Modules/restapi/rest_manager.py

```diff
@@ -24,10 +27,42 @@ def start(self):
         Starts the HTTP API with the listen port as specified in the session configuration.
         """
         self.site = reactor.listenTCP(self.session.get_http_api_port(),
-                                      server.Site(RootEndpoint(self.session)))
+                                      server.Site(resource=RootEndpoint(self.session),
+                                                  requestFactory=RESTRequest))
 
     def stop(self):
         """
         Stop the HTTP API and return a deferred that fires when the server has shut down.
         """
         return maybeDeferred(self.site.stopListening)
+
+
+class RESTRequest(server.Request):
+    """
+    This class gracefully takes care of unhandled exceptions raised during the processing of any request.
+    """
+    defaultContentType = b"text/json"
```

Comment:
@lfdversluis @devos50 after this commit is merged, all the setcontent type calls can be removed from all endpoints